packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/types-section.tsx

```diff
@@ -1,59 +1,119 @@
-import { Box, Typography } from "@mui/material";
+import { Box } from "@mui/material";
+import { useEffect, useState } from "react";
 
+import { useBlockProtocolUpdateEntity } from "../../../../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
+import { useBlockProtocolAggregateEntityTypes } from "../../../../../components/hooks/block-protocol-functions/ontology/use-block-protocol-aggregate-entity-types";
 import { SectionWrapper } from "../../../shared/section-wrapper";
-import { WhiteCard } from "../../../shared/white-card";
 import { useEntityEditor } from "./entity-editor-context";
-
-interface TypeCardProps {
-  url: string;
-  title: string;
-}
-
-const TypeCard = ({ url, title }: TypeCardProps) => {
-  return (
-    <WhiteCard href={url}>
-      <Box
-        sx={{
-          display: "flex",
-          alignItems: "center",
-          px: 1.5,
-          py: 1.25,
-          gap: 1.25,
-          color: ({ palette }) => palette.black,
-        }}
-      >
-        <FontAwesomeIcon icon={faAsterisk} />
-        <Typography variant="smallTextLabels" fontWeight={600}>
-          {title}
-        </Typography>
-      </Box>
-    </WhiteCard>
-  );
-};
+import { EntityVersionUpdateModal } from "./types-section/entity-version-update-moda";
+import { TypeCard } from "./types-section/type-card";
 
 export const TypesSection = () => {
-  const { entitySubgraph } = useEntityEditor();
+  const { entitySubgraph, refetch } = useEntityEditor();
 
   const entity = getRoots(entitySubgraph)[0]!;
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+  const {
+    metadata: { editionId, entityTypeId },
+    properties,
+  } = entity;
+
+  const { aggregateEntityTypes } = useBlockProtocolAggregateEntityTypes();
+  const [newVersion, setNewVersion] = useState<number>();
+  const [updateModalOpen, setUpdateModalOpen] = useState(false);
+  const [updatingVersion, setUpdatingVersion] = useState(false);
+
+  useEffect(() => {
+    const init = async () => {
+      const res = await aggregateEntityTypes({ data: {} });
+
+      const entityTypeWithSameBaseId = res.data?.roots.find(
+        (root) => root.baseId === extractBaseUri(entityTypeId),
+      );
+
+      if (!entityTypeWithSameBaseId) {
+        return;
+      }
 
-  const entityTypeTitle = getEntityTypeById(
-    entitySubgraph,
-    entity.metadata.entityTypeId,
-  )!.schema.title;
-  const entityTypeUrl = entity.metadata.entityTypeId.replace(/v\/\d+/, "");
+      const currentEntityVersion = extractVersion(entityTypeId);
+
+      if (entityTypeWithSameBaseId.version > currentEntityVersion) {
+        setNewVersion(entityTypeWithSameBaseId.version);
+      }
+    };
+
+    void init();
+  }, [aggregateEntityTypes, entityTypeId]);
+
+  const entityType = getEntityTypeById(entitySubgraph, entityTypeId);
+  const entityTypeTitle = entityType?.schema.title ?? "";
+  const entityTypeUrl = extractBaseUri(entityTypeId);
```

Comment:
```suggestion
  const entityTypeBaseUri = extractBaseUri(entityTypeId);
```

It's quite strange to just call it "Url" it's not very clear _which_ URL you're on about

Reply:
Updated here https://github.com/hashintel/hash/pull/1774/commits/568780a6ff5c91edc987c91d71f78348d97a51d0